src/Tickets/Commerce/Gateways/Stripe/Payment_Intent_Handler.php

```diff
@@ -0,0 +1,206 @@
+				__( 'There was an error enabling Stripe on your cart. More information is available in the Event Tickets settings dashboard. Please contact the site administrator for support.', 'event-tickets' ),
+			];
+		}
+
+		return $this->store_payment_intent( $payment_intent );
+	}
+
+	/**
+	 * Updates an existing payment intent to add any necessary data before confirming the purchase.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $data the purchase data received from the front-end
+	 *
+	 * @return array|\WP_Error|null
+	 */
+	public function update_payment_intent( $data ) {
+		$payment_intent_id = $data['payment_intent']['id'];
+
+		$stripe_receipt_emails = tribe_get_option( Settings::$option_stripe_receipt_emails );
+
+		// Currently this method is only used to add an email recipient for Stripe receipts. If this is not
+		// required, only return the payment intent object to store.
+		if ( ! $stripe_receipt_emails ) {
+			return Payment_Intent::get( $payment_intent_id );
+		}
+
+		if ( $stripe_receipt_emails && ! empty( $data['billing_details']['email'] ) ) {
+			$body['receipt_email'] = $data['billing_details']['email'];
+		}
+
+		return Payment_Intent::update( $payment_intent_id, $body );
+	}
+
+	/**
+	 * Assembles basic data about the payment intent created at page-load to use in javascript
+	 *
+	 * @since TBD
+	 *
+	 * @return array
+	 */
+	public function get_publishable_payment_intent_data() {
+		$pi = $this->get_payment_intent_transient();
+
+		if ( empty( $pi ) ) {
+			return [];
+		}
+
+		if ( ! empty( $pi['errors'] ) ) {
+			return $pi;
+		}
+
+		return [
+			'id'   => $pi['id'],
+			'key'  => $pi['client_secret'],
+			'name' => $this->get_payment_intent_transient_name(),
+		];
+	}
+
+	/**
+	 * Compose the transient name used for payment intent transients
+	 *
+	 * @since TBD
+	 */
+	public function set_payment_intent_transient_name() {
+		$this->payment_intent_transient_name = $this->payment_intent_transient_prefix . md5( tribe( Cart::class )->get_cart_hash() );
+	}
+
+	/**
+	 * Returns the transient name used for payment intent transients
+	 *
+	 * @since TBD
+	 *
+	 * @return string
+	 */
+	public function get_payment_intent_transient_name() {
+
+		if ( empty( $this->payment_intent_transient_name ) ) {
+			$this->set_payment_intent_transient_name();
+		}
+
+		return $this->payment_intent_transient_name;
+	}
+
+	/**
+	 * Retrieve a stored payment intent referring to the current cart
+	 *
+	 * @since TBD
+	 *
+	 * @return array|false
+	 */
+	public function get_payment_intent_transient() {
+		return get_transient( $this->get_payment_intent_transient_name() );
+	}
+
+	/**
+	 * Delete the payment intent transient
+	 *
+	 * @since TBD
+	 *
+	 * @return bool
+	 */
+	public function delete_payment_intent_transient() {
+		return delete_transient( $this->get_payment_intent_transient_name() );
+	}
+
+	/**
+	 * Store a payment intent array in a transient
+	 *
+	 * @since TBD
+	 *
+	 * @param array $payment_intent payment intent data from Stripe
```

Comment:
```suggestion
	 * Store a payment intent array in a transient.
	 *
	 * @since TBD
	 *
	 * @param array $payment_intent Payment intent data from Stripe.
```